tests/LifeCycleSpec.hs

```diff
@@ -0,0 +1,34 @@
+
+{-# OPTIONS_GHC -Wall -Werror #-}
+
+module LifeCycleSpec
+where
+
+import Data.List (intersect)
+import Test.Hspec (Spec, it)
+import Test.QuickCheck.Property
+
+import Arbitrary
+import AulaTests (TestSuite(..), tag)
+import LifeCycle
+
+
+spec :: Spec
+spec = do
+    tag Large . it "sameItemDoesNotAppear as clickable and grayed out in ideaCapabilities" . property .
+        forAllShrinkDef arb $ \userId ->
+        forAllShrinkDef arb $ \role ->
+        forAllShrinkDef arb $ \idea ->
+        forAllShrinkDef arb $ \phase ->
```

Comment:
you can probably make this one function in a 4-tuple?  if you want to, you can define `curry4` in `Types` to make this more compact.  (-: